### PR TITLE
ITKDev - patron dashboard

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

https://platform.dandigbib.org/issues/5374

#### Description

Add dashboard module 

**Remember to revert branches!!**

#### Screenshot of the result

n/a

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

**Remember** to revert design and react in composer.json before merge

### CODE REVIEW

File: web/modules/custom/dpl_dashboard/src/Plugin/Block/DashboardBlock.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      // Urls.
+      'dpl-cms-base-url' => DplReactAppsController::dplCmsBaseUrl(),
+      'fees-page-url' => $generalSettings->get('fees_page_url'),
+      'intermediate-url' => $dashboardSettings->get(""),
+      'loans-not-overdue-url' => $dashboardSettings->get(""),
+      'loans-overdue-url' => $dashboardSettings->get(""),
+      'loans-soon-overdue-url' => $dashboardSettings->get(""),
+      'pay-owed-url' => $dashboardSettings->get(""),
+      'physical-loans-url' => $dashboardSettings->get(""),
+      'reservations-url' => $dashboardSettings->get(""),
```
Comment (cableman) [inline web/modules/custom/dpl_dashboard/src/Plugin/Block/DashboardBlock.php:86]: There are a lot of empty configuration getters here?

**Review by cableman — CHANGES_REQUESTED** [review]

File: web/modules/custom/dpl_dashboard/src/Plugin/Block/DashboardBlock.php (hunk 1 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_dashboard/src/Plugin/Block/DashboardBlock.php:86]: added

File: composer.json [hunk 2]
```diff
@@ -17,10 +17,10 @@
             "type": "package",
             "package": {
                 "name": "danskernesdigitalebibliotek/dpl-react",
-                "version": "0.3.0-rc28",
+                "version": "0.3.0-rc40",
                 "type": "drupal-library",
                 "dist": {
-                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-react/releases/download/release-develop/dist.zip",
+                    "url": "https://github.com/itk-dev/dpl-react/releases/download/release-fix%2Fremoving-zeros-from-menu/dist.zip",
```
Comment (spaceo) [inline composer.json:23]: remember to use develop

File: composer.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
                 "type": "drupal-library",
                 "version": "v1.7.0-rc19",
                 "dist": {
-                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-design-system/releases/download/release-develop/dist.zip",
+                    "url": "https://github.com/itk-dev/dpl-design-system/releases/download/release-feature%2F5374-dashboard-page-styling/dist.zip",
```
Comment (spaceo) [inline composer.json:38]: remember to use develop

File: web/modules/custom/dpl_dashboard/src/Controller/DplDashboardController.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+   */
+  public function list(): array {
+    /** @var \Drupal\dpl_dashboard\Plugin\Block\DashboardBlock $plugin_block */
+    $plugin_block = $this->blockManager->createInstance('dpl_dashboard_block', []);
+
+    // Some blocks might implement access check.
+    $access_result = $plugin_block->access($this->currentUser());
+    if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+      throw new AccessDeniedHttpException();
+    }
```
Comment (spaceo) [inline web/modules/custom/dpl_dashboard/src/Controller/DplDashboardController.php:61]: Optionally add a TODO about using a service or other common Drupal permission functionality in the future

**Review by spaceo — APPROVED** [review]
Approved. But I have some comments

File: web/modules/custom/dpl_dashboard/src/Controller/DplDashboardController.php (hunk 4 above)
Comment (cableman) [reply web/modules/custom/dpl_dashboard/src/Controller/DplDashboardController.php:62]: Ticket create about this, so we remember this optimization: https://platform.dandigbib.org/issues/5801

File: composer.json (hunk 3 above)
Comment (cableman) [reply composer.json]: Fixed

File: composer.json (hunk 2 above)
Comment (cableman) [reply composer.json]: Fixed

**Review by cableman — APPROVED** [review]

**Pull request merged**